### PR TITLE
Feature/flagged nodes confirm ham [ENG-488]

### PR DESCRIPTION
**Pull request by corbinSanders**
## Purpose

Break out the "Confirm Ham"/"Confirm Spam" buttons and modal dialogs into a template. Add in capability for bulk Confirm Ham operations. 

## Changes

- Modified admin/nodes.py NodeFlaggedSpamList to add functionality for bulk HAM operations.
- Created admin/templates/nodes/ham_spam_modal.html template for Ham/Spam button and modal.
- Modified admin/templates/nodes/node_list.html to utilize the ham_spam_modal.html template.

## QA Notes

Test /nodes/flagged_spam to ensure Confirm Ham and Confirm Spam is working properly.

## Ticket

https://openscience.atlassian.net/browse/ENG-488

### CODE REVIEW

File: admin/nodes/views.py [hunk 1]
```diff
@@ -349,23 +349,34 @@ class NodeFlaggedSpamList(NodeSpamList, DeleteView):
...
+                )
+            else:
+                node.confirm_ham(save=True)
+                update_admin_log(
+                    user_id=self.request.user.id,
+                    object_id=nid,
+                    object_repr='Node',
+                    message='Confirmed HAM: {}'.format(nid),
+                    action_flag=CONFIRM_HAM
+                )
```
Comment (pattisdr) [inline admin/nodes/views.py]: To future-proof this, I don't know if we should assume that if `spam_confirm` is not in request.POST.keys(), mark everything as ham.  I'd do an if/elif instead.

Comment (corbinSanders) [reply admin/nodes/views.py]: I agree. The elif makes it more explicit and readable as well.

**Comment by pattisdr** [comment]
LGTM ⛵️

File: admin/nodes/views.py [hunk 2]
```diff
@@ -349,23 +349,34 @@ class NodeFlaggedSpamList(NodeSpamList, DeleteView):
     template_name = 'nodes/flagged_spam_list.html'
 
     def delete(self, request, *args, **kwargs):
-        if not request.user.has_perm('auth.mark_spam'):
+        if (('spam_confirm' in request.POST.keys() and not request.user.has_perm('auth.mark_spam')) or
```
Comment (Johnetordoff) [inline admin/nodes/views.py]: Another future proof thing is to wrap any dict iterator in a list like `list(request.POST.keys())` or `list(request.POST.items())`. This is compatible with both Python 2 and 3

Comment (pattisdr) [reply admin/nodes/views.py]: ah ok, @Johnetordoff, looking at [PEP 3106: Dictionary Views](https://docs.python.org/dev/whatsnew/2.7.html#pep-3106-dictionary-views), will this be required to work with Py3?

Comment (Johnetordoff) [reply admin/nodes/views.py]: yes it's already part of the WIP PR, just might as well wrap these in new PRs before we are required to as part of Py3.

Comment (pattisdr) [reply admin/nodes/views.py]: great, John, thanks for pointing this out!

**Pull request merged**